internal/cmdsearch/putpattern_test.go

```diff
@@ -89,6 +89,38 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: dev/my-project/nginx # kpt-set: ${env}/${project}/${name}
+spec:
+  replicas: 3
+ `,
+	},
+	{
+		name: "put pattern by regex",
```

Comment:
More test cases, in particular test what happens if we can't resolve the setter value from the pattern.

Reply:
This is more of a high level test. Regression tests are in search_test.go with different cases.

Reply:
Yeah, but those doesn't really show/test what happens from the users point of view, so it would be useful to demonstrate an error case in this test.

Reply:
Added a test-case.